Mathlib/Computability/Language.lean

```diff
@@ -215,12 +215,8 @@ theorem mem_pow {l : Language α} {x : List α} {n : ℕ} :
 
 theorem kstar_eq_iSup_pow (l : Language α) : l∗ = ⨆ i : ℕ, l ^ i := by
   ext x
-  simp only [mem_kstar, mem_iSup, mem_pow]
-  constructor
-  · rintro ⟨S, rfl, hS⟩
-    exact ⟨_, S, rfl, rfl, hS⟩
-  · rintro ⟨_, S, rfl, rfl, hS⟩
-    exact ⟨S, rfl, hS⟩
+  simp_rw [mem_kstar, mem_iSup, mem_pow]
+  tauto
```

Comment:
```
theorem kstar_eq_iSup_pow (l : Language α) : l∗ = ⨆ i : ℕ, l ^ i := by
  aesop (add simp [mem_iSup, mem_pow, mem_kstar])
```
automation can do the whole thing, with a tiny bit of help :)

Reply:
There is a noticeable difference in performance when using `aesop` - despite that, should I commit it like this? (I'm under the impression that `tauto` is fast, which is why I made the current golf.)

Reply:
I think either is fine, up to you. In my opinion `ext; simp [mem_kstar, mem_iSup, mem_pow]` should be able to do it, and it's a deficiency in `simp` that it can't

Reply:
That makes sense. I committed the `aesop` golf, but a quick investigation into the proof solved by `tauto` shows a useless exists quantifier 
![A screenshot of a lean golf view depicting `(∃ L, x✝ = L.flatten ∧ ∀ y ∈ L, y ∈ l) ↔ ∃ i S, x✝ = S.flatten ∧ S.length = i ∧ ∀ y ∈ S, y ∈ l`, where the `i` in `∃ i` is hovered as having a type of the natural numbers](https://github.com/user-attachments/assets/ac7c7f88-7b89-48ad-b1e9-1e7839e564b3).

There could be an extra lemma that is a special form of the combination of `mem_iSup` and `mem_pow` that removes the extra exists quantifier there, but from what I know, `tauto`/`aesop` would both be good tactics here

Reply:
Right, the general pattern is mentioned here: [#mathlib4 > potential simp_procs @ 💬](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/potential.20simp_procs/near/506852714), and an upgrade to that simproc (which I had thought was the existing behaviour already) would make this work

Reply:
I'll close this PR in favor of making that simp proc more powerful - I'll see if I can make that today.